src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/VmrDiffOperation.cs

```diff
@@ -0,0 +1,343 @@
+                "clone",
+                "--depth", "1",
+                repo.Path,
+                "-b", repo.Branch,
+                tmpProductRepo
+            ]);
+        }
+        else
+        {
+            await CheckoutBranch(repo);
+            CopyDirectory(repo.Path, tmpProductRepo, true);
+        }
+
+        return tmpProductRepo;
+    }
+
+    private async Task<(DiffRepo repo1, DiffRepo repo2)> ParseInput()
+    {
+        DiffRepo repo1, repo2;
+        var parts = options.Input.Split("..", StringSplitOptions.RemoveEmptyEntries);
+        if (parts.Length > 2 || parts.Length < 1)
+        {
+            throw new ArgumentException($"Invalid input {options.Input}");
+        }
+        
+        if (parts.Length == 1)
+        {
+            var currentPath = Directory.GetCurrentDirectory();
+            var res = await processManager.ExecuteGit(currentPath, "status");
+            res.ThrowIfFailed("Current directory is not a git repo");
+            repo1 = new DiffRepo(currentPath, string.Empty, true, IsLocalRepoVmr(currentPath));
+            repo2 = await ParseRepo(parts[0]);
+        }
+        else
+        {
+            repo1 = await ParseRepo(parts[0]);
+            repo2 = await ParseRepo(parts[1]);
+        }
+        await VerifyInput(repo1, repo2);
+
+        return (repo1, repo2);
+    }
+
+    private async Task<IReadOnlyCollection<string>> GetDiffFilters(string mapping)
+    {
+        var gitRepo = gitRepoFactory.CreateClient(DarcLib.Constants.DefaultVmrUri);
+        return sourceMappingParser.ParseMappingsFromJson(
+            (await gitRepo.GetFileContentsAsync($"{VmrInfo.SourceDirName}/{VmrInfo.SourceMappingsFileName}", DarcLib.Constants.DefaultVmrUri, "main")))
+            .Where(m => m.Name == mapping)
+            .First().Exclude;
+    }
+
+    private async Task CheckoutBranch(DiffRepo repo)
+    {
+        if (!string.IsNullOrEmpty(repo.Branch))
+        {
+            var res = await processManager.ExecuteGit(repo.Path, [
+                    "checkout",
+                    repo.Branch
+                ]);
+            res.ThrowIfFailed($"Failed to checkout requested branch {repo.Branch} in {repo.Path}");
+        }
+    }
+
+    private async Task<DiffRepo> ParseRepo(string input)
+    {
+        var branchIndex = input.LastIndexOf(':');
+        if (branchIndex == -1)
+        {
+            throw new ArgumentException("Invalid input format. Expected format is 'repo:branch'");
+        }
+        var repo = input.Substring(0, branchIndex);
+        var branch = input.Substring(branchIndex + 1);
+        bool isLocal = !repo.StartsWith(HttpsPrefix);
+        bool isVmr = isLocal ? IsLocalRepoVmr(repo) : await IsRemoteRepoVmr(repo, branch);
+        return new DiffRepo(repo, branch, isLocal, isVmr);
+    }
+
+    private async Task VerifyInput(DiffRepo repo1, DiffRepo repo2)
+    {
+        if (!(repo1.IsVmr ^ repo2.IsVmr))
```

Comment:
Not sure which is better but this could also be 
```suggestion
        if (repo1.IsVmr == repo2.IsVmr)
```

Reply:
`==` is shorter. I really wanted xor tho ahah